prov/gni/src/gnix_av.c

```diff
@@ -446,12 +446,12 @@ static int map_insert(struct gnix_fid_av *int_av, const void *addr,
 
 		((struct gnix_address *)fi_addr)[i] = temp.gnix_addr;
 		the_entry =  &blk->base[i];
-		memcpy(&the_entry->gnix_addr, &temp.gnix_addr,
-		       sizeof(struct gnix_address));
+		the_entry->gnix_addr.device_addr = temp.gnix_addr.device_addr;
+		the_entry->gnix_addr.cdm_id = temp.gnix_addr.cdm_id;
```

Comment:
Agreed.  I will update this PR with a macro.
> On May 9, 2016, at 2:37 AM, Howard Pritchard <notifications@github.com> wrote:
> 
> In prov/gni/src/gnix_av.c <https://github.com/ofi-cray/libfabric-cray/pull/812#discussion_r62458871>:
> 
> > @@ -446,12 +446,12 @@ static int map_insert(struct gnix_fid_av *int_av, const void *addr,
> >  
> >  		((struct gnix_address *)fi_addr)[i] = temp.gnix_addr;
> >  		the_entry =  &blk->base[i];
> > -		memcpy(&the_entry->gnix_addr, &temp.gnix_addr,
> > -		       sizeof(struct gnix_address));
> > +		the_entry->gnix_addr.device_addr = temp.gnix_addr.device_addr;
> > +		the_entry->gnix_addr.cdm_id = temp.gnix_addr.cdm_id;
> defining a macro to copy a gnix_addr would be nice for these changes.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/ofi-cray/libfabric-cray/pull/812/files/abc41c6cb76334b1f6e1fdc1a24af7a70c343be6#r62458871>